### ISSUE TITLE
Idea: Provide an embeddable frame of event data for use in other websites

### ISSUE THREAD

**Issue by tytremblay** [the-blue-alliance/the-blue-alliance#1262]
Allow teams to embed the information below as a frame with internal scrolling on other websites.

![image](https://cloud.githubusercontent.com/assets/6652493/10620347/90f8da7c-7748-11e5-96e9-785020496733.png)

**Comment by fangeugene** [comment]
Wondering what the best way to do this is.

The simplest is an iframe embed, but that would offload all rendering and bandwidth on us.

Another is a JS widget that hits the API to display the info. Since the API is heavily cached, this may be a better option. This option can already be implemented by third parties if they want to, but having a provided widget can standardize the look and perhaps brand it with "Provided by The Blue Alliance" or something.

**Comment by gregmarra** [comment]
We can also cache the iframe html

**Comment by tytremblay** [comment]
Does that mean this is doable? NE FIRST would like to use this on their new site.

**Comment by phil-lopreiato** [comment]
@tytremblay yeah, it should be pretty doable.

Maybe we could structure the client side stuff similar to the [FB Page Plugin](https://developers.facebook.com/docs/plugins/page-plugin), where you just have to include a script + add a div with some data elements to configure what you see. 

As long as we can get the js file cached on a CDN and make sure all API requests hit cache (which they already do), it shouldn't be a burden at all

**Comment by tytremblay** [comment]
Excellent. Is it too early to ask about lead time?  Any way I can help?

**Comment by phil-lopreiato** [comment]
@fangeugene how do you want to run this? I'm partial to your second idea, the js widget (iframes suck and we can rely on API caching and stuff. We can also leverage firebase to post to an event-specific feed, so we can just render a [notification ticker](http://tba-notification-firehose.appspot.com/) client side, like on GameDay

@tytremblay what do you want it to look like? I'm not sure just a cut/paste of the event page would translate well to a widget. I was thinking of something like this...
![](http://imgur.com/XDAi3qJ.jpg)
If you want to write some client side stuff, you should just have to hit the Event Info/Event Teams endpoints, plus some JS for firebase. I can direct you to places in the TBA repo, if you want

**Comment by tytremblay** [comment]
NE FIRST is thinking something like this:http://i.imgur.com/8Nv3BDF.png

**Comment by tytremblay** [comment]
The idea is to reduce the amount of variability between event pages in NE. Id like users to have a unified experience of team lists, scores, rankings, awards, etc. 

I think it would also spread the TBA word to parents and teams that aren't full time users yet.

**Comment by phil-lopreiato** [comment]
Okay, I see what you mean. So then a copy/paste of the TBA event page would fit in, the Event Controller should adapt pretty simply. 

I also think it would be a cool idea to have an embeddable ticker for event notifications down the right column under the navigation.

**Comment by tytremblay** [comment]
Agreed. That'd be the dream. If we could plug all NE events into the ticker widget and let it churn away for the season I think it would add a lot of value.

**Comment by phil-lopreiato** [comment]
I added per-event and per-district firebase feeds in 668abd2184023b12316da71c6015ac403143d145, so we'd just have to include some client-side js to render the ticker nicely.

@fangeugene @gregmarra what do you think the best way to do the widget parts would be?

**Comment by minecraftchest1** [comment]
Any updates on this?
